src/app/models/submission.server.model.ts

```diff
@@ -255,42 +256,82 @@ EncryptSubmissionSchema.statics.findSingleMetadata = function (
   formId: string,
   submissionId: string,
 ): Promise<StorageModeSubmissionMetadata | null> {
-  return (
-    this.findOne(
-      {
-        form: formId,
-        _id: submissionId,
+  const pageResults: Promise<MetadataAggregateResult[]> = this.aggregate([
+    {
+      $match: {
+        form: mongoose.Types.ObjectId(formId),
+        _id: mongoose.Types.ObjectId(submissionId),
         submissionType: SubmissionType.Encrypt,
       },
-      { created: 1 },
-    )
-      // Reading from primary to avoid any contention issues with bulk queries
-      // on secondary servers.
-      .read('primary')
-      .then((result) => {
-        if (!result) {
-          return null
-        }
-
-        // Build submissionMetadata object.
-        const metadata: StorageModeSubmissionMetadata = {
-          number: 1,
-          refNo: result._id,
-          submissionTime: moment(result.created)
-            .tz('Asia/Singapore')
-            .format('Do MMM YYYY, h:mm:ss a'),
-        }
-
-        return metadata
-      })
-  )
+    },
+    {
+      $lookup: {
+        from: 'payments',
+        localField: 'paymentId',
+        foreignField: '_id',
+        as: 'payments',
+      },
+    },
+    {
+      $project: {
+        _id: 1,
+        created: 1,
+        'payments.payout': 1,
+        'payments.completedPayment': 1,
+        'payments.amount': 1,
+        'payments.email': 1,
+      },
+    },
+  ])
+    .limit(1)
+    .exec()
+
+  return Promise.resolve(pageResults).then((results) => {
+    if (!results || results.length <= 0) {
+      return null
+    }
+
+    const result = results[0]
+    const paymentMeta = result.payments?.[0]
+
+    // Build submissionMetadata object.
+    const metadata: StorageModeSubmissionMetadata = {
+      number: 1,
+      refNo: result._id,
+      submissionTime: moment(result.created)
+        .tz('Asia/Singapore')
+        .format('Do MMM YYYY, h:mm:ss a'),
+      payments: paymentMeta
```

Comment:
likewise for meta, we can add a function to build paymentMeta, so that it can be used across both functions

Reply:
Great suggestion. Added in f80f726a2